### ISSUE TITLE
ESM plugins do not work with begin dev

### ISSUE THREAD

**Issue by ryanbethel** [beginner-corp/cli#154]
ESM plugins don't work with the Begin CLI using `begin dev`. They do, however, work when using `sandbox` directly.

Steps to reproduce: 
Using @brianleroux 's S3 upload repo at [https://github.com/brianleroux/enhance-example-s3-upload](https://github.com/brianleroux/enhance-example-s3-upload) . 
1. clone the repo `git clone https://github.com/brianleroux/enhance-example-s3-upload.git`
2. install deps `npm i`
3. run `begin dev`
4. install sandbox `npm i @architect/sandbox`
5. run sandbox `npx sandbox`

You should see the following error when running `begin dev` but no error running `npx sandbox`:
```sh
Error: Plugin error:
- Unable to load plugin 's3': Invalid host defined options
    at getPluginModules (/snapshot/cli/node_modules/@architect/inventory/src/config/pragmas/plugins.js)
    at getPluginModulesCallbackified (node:util:306:5)
    at architectInventory (/snapshot/cli/node_modules/@architect/inventory/src/index.js)
    at action (/snapshot/cli/src/commands/dev/index.js)
    at runCommand (/snapshot/cli/src/commands/index.js)
    at begin (/snapshot/cli/src/index.js)
    at Object.<anonymous> (/snapshot/cli/src/index.js)
    at Module._compile (pkg/prelude/bootstrap.js:1926:22)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
➜  enhance-example-s3-upload git:(main) ✗
```

**Comment by ryanblock** [comment]
Under the hood it is just an import statement. As such, paths need to include file names.

**Comment by brianleroux** [comment]
Let's document this

**Comment by ryanblock** [comment]
Uh oh: https://stackoverflow.com/questions/69734757/dynamic-import-import-fails-when-the-code-is-packaged-into-an-executable

> Okay, for some reason node executables made with package [pkg](https://www.npmjs.com/package/pkg) doesn't honor import() statements.

**Comment by ryanblock** [comment]
See also: https://github.com/vercel/pkg/issues/1291

**Comment by brianleroux** [comment]
What a nightmare.

**Comment by ryanblock** [comment]
Indeed, apologies for the early closure of the issue @ryanbethel, Brian and I had worked through another esm plugin issue (the file name) from this repo already, I assumed it was the same.

So I think the answer here for now is: Begin CLI currently does not support ESM plugins (due to limitations in pkg). Thoughts?

**Comment by macdonst** [comment]
It's frustrating that we are in a situation with `pkg` that `begin dev` does not have parity with `arc sandbox`. From a DX perspective, I'd expect this to "just work" in `begin dev`. We already have an issue explaining the difference between Architect, Enhance and Begin and this further confuses things. 

**Note:** this is me being frustrated with `pkg`.

**Comment by brianleroux** [comment]
its a deep sunk cost discussion but i feel like we should embrace `npm i -g @begin/cli` and walk on this approach for now

**Comment by ryanblock** [comment]
Added better error handling when ESM plugins fail in pkg-compiled binary distributions; it's not a real fix, but it's at least a bit less bewildering!
